src/pykeen/nn/emb.py

```diff
@@ -118,6 +118,47 @@ def get_in_canonical_shape(
             x = x.unsqueeze(dim=1)
         return x
 
+    def get_in_more_canonical_shape(
+        self,
+        dim: Union[int, str],
+        indices: Optional[torch.LongTensor] = None,
+    ) -> torch.FloatTensor:
+        """Get representations in canonical shape.
+
+        The canonical shape is given as
+
+        (batch_size, d_1, d_2, d_3, ``*``)
+
+        fulfilling the following properties:
+
+        Let i = dim. If indices is None, the return shape is (1, d_1, d_2, d_3) with d_i = num_representations,
```

Comment:
Maybe add a concrete example?

Reply:
@mberr it wouldn't hurt to have some unit tests for this, but I never found any in #107 to port here

Reply:
> Maybe add a concrete example?

done in 84648c9441b58834aec8b07c53a19d0a7efea5e2

Reply:
> @mberr it wouldn't hurt to have some unit tests for this, but I never found any in #107 to port here

done in 982df01c0d2bddca8fe92bb6fb105ecb3d738c62